File: backend/services/language_detector.py
import re

class LanguageDetector:
    def __init__(self):
        self.hinglish_keywords = {
            'bhai', 'hain', 'kya', 'chahiye', 'mein', 'kitna', 'kaise', 'milega',
            'kab', 'aayega', 'namaste', 'mujhe', 'aaj', 'par', 'hi', 'batao',
            'sasta', 'mehenga', 'discount', 'size', 'bhi', 'na', 'wala', 'wali'
        }

    def detect_language(self, text: str) -> str:
        if not text or not isinstance(text, str):
            return 'english'

        # Check Devanagari script
        if re.search(r'[\u0900-\u097F]', text):
            return 'hindi'

        words = re.sub(r'[^a-z0-9\s]', '', text.lower()).split()
        match_count = sum(1 for w in words if w in self.hinglish_keywords)

        if match_count >= 1:
            return 'hinglish'

        return 'english'

language_detector = LanguageDetector()

File: backend/src/services/languageDetector.ts
export class LanguageDetector {
  private hinglishKeywords = [
    'bhai', 'hain', 'kya', 'chahiye', 'mein', 'kitna', 'kaise', 'milega',
    'kab', 'aayega', 'namaste', 'mujhe', 'aaj', 'par', 'hi', 'batao',
    'sasta', 'mehenga', 'discount', 'size', 'bhi', 'na', 'wala', 'wali'
  ];

  public detectLanguage(text: string): 'english' | 'hindi' | 'hinglish' {
    if (!text || typeof text !== 'string') return 'english';

    // Check for Devanagari Unicode Range
    if (/[\u0900-\u097F]/.test(text)) {
      return 'hindi';
    }

    const lower = text.toLowerCase();
    const words = lower.replace(/[^a-z0-9\s]/g, '').split(/\s+/);

    let matchCount = 0;
    for (const w of words) {
      if (this.hinglishKeywords.includes(w)) {
        matchCount++;
      }
    }

    if (matchCount >= 1) {
      return 'hinglish';
    }

    return 'english';
  }
}

export const languageDetector = new LanguageDetector();
